app/src/main-process/main.ts

```diff
@@ -450,14 +451,24 @@ app.on('ready', () => {
    * Handle the action to show a contextual menu.
    *
    * It responds an array of indices that maps to the path to reach
-   * the menu (or submenu) item that was clicked or null if the menu
-   * was closed without clicking on any item.
+   * the menu (or submenu) item that was clicked or null if the menu was closed
+   * without clicking on any item or the item click was handled by the main
+   * process as opposed to the renderer.
    */
-  ipcMain.handle('show-contextual-menu', (event, items) => {
-    return new Promise(resolve => {
-      const menu = buildContextMenu(items, indices => resolve(indices))
+  ipcMain.handle('show-contextual-menu', (event, items, addSpellCheckMenu) => {
+    return new Promise(async resolve => {
       const window = BrowserWindow.fromWebContents(event.sender) || undefined
 
+      const spellCheckMenuItems = addSpellCheckMenu
```

Comment:
Something I wasn't sure of with this approach is that the buildContextMenu takes a call back with the menu items indices so it can send back to the renderer which item was clicked (and that is how spell check menu items were invoked previously). However, if we send it items with roles like edit-menu, it adds menu items so as cut, copy, paste and when those are clicked null is returned. I was unsure if the spell check menu items had to specifically have a call back sending back null send the renderer is no longer responsible for invoke thing action, but it appears they are handled just as if you are clicking off the menu or one of the electron role based menu items and does resolve(null) - I am assuming that is the `menu.popoup... resolve(null)` being invoked.